.travis.yml

```diff
@@ -0,0 +1,57 @@
+language: c
+dist: trusty   # Travis build machine is Ubuntu Trusty 14.04 x64
+sudo: required # A sudo enabled, full VM is used for building instead of a docker container
+
+git:
+  submodules: false
+
+before_install:
+  - sudo apt-get update -qq
+  # Pre-reqs for building dsc
+  - sudo apt-get install -qq pkg-config make bison g++ rpm librpm-dev libpam0g-dev libssl-dev libmysqlclient-dev 
+
+# Travis-CI automatically clones the Powershell-DSC-for-Linux repo since this is where this .travis.yml file is located
+# We will ignore this clone and clone the superproject Build-Powershell-DSC-for-Linux as follows
+
+install:
+  - cd ..
+  - git clone https://github.com/Microsoft/Build-Powershell-DSC-for-Linux.git ./Build-Powershell-DSC-for-Linux
+  - cd ./Build-Powershell-DSC-for-Linux
+  # change url from ssh to https for submodules, travis-ci cannot download a submodule that is on a public url.
+  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules 
+  - git submodule update --init --recursive dsc omi pal
+  - # Do the same for ostc-openssl
+  - cd ..
+  - git clone https://github.com/Microsoft/ostc-openssl.git ./ostc-openssl
+  - cd ./ostc-openssl
+  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules 
+  - git submodule update --init --recursive openssl-0.9.8 openssl-1.0.0
+  - cd ../Build-Powershell-DSC-for-Linux
+
+before_script:
+  - git checkout master && git submodule foreach git checkout master
+  - cd dsc
+  # PR validation during check-in will be setup in github in the settings section of our repo
+  - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then git fetch origin refs/pull/$TRAVIS_PULL_REQUEST/head ; git checkout FETCH_HEAD; fi 
+  - cd build
+  # pwd: /home/travis/build/Microsoft/Build-Powershell-DSC-for-Linux/dsc/build
+  # Since 'build' is a keyword for travis-CI, change Makefile to substitute 'dsc/build' instead of 'build'
+  - sed -i 's/(subst \/build,,$(CURDIR))/(subst dsc\/build,dsc,$(CURDIR))/' Makefile 
```

Comment:
I hit this too, and I actually fixed the makefile itself to not hit this (used a different mechanism). That's much cleaner than actually modifying your Makefile each time for CI, I think.